projects/dataset_recorder/data_recorder.cc

```diff
@@ -0,0 +1,246 @@
+// standard includes
+#include<string>
+#include<vector>
+#include <thread>
+#include <mutex>
+#include <atomic>
+
+// BoB includes
+#include "common/main.h"
+#include "common/stopwatch.h"
+#include "common/background_exception_catcher.h"
+#include "common/gps.h"
+#include "common/map_coordinate.h"
+#include "common/bn055_imu.h"
+
+#include "imgproc/opencv_unwrap_360.h"
+#include "video/panoramic.h"
+
+#include <sys/stat.h>
+#include <ctime>
+
+
+
+
+
+
+// Standard C++ includes
+#include <iostream>
+
+using namespace BoBRobotics;
+using namespace BoBRobotics::ImgProc;
+using namespace BoBRobotics::Video;
+
+
+#define WIDTH 192
+#define HEIGHT 72
+
+#define IS_UNWRAP true
+#define UPDATE_INTERVAL 200
+
+
+void readCameraThreadFunc(BoBRobotics::Video::Input *cam,
+                        cv::Mat &img,
+                        std::atomic<bool> &shouldQuit,
+                        std::mutex &mutex)
+{
+
+    //auto unwrapper = cam->createUnwrapper(cv::Size(WIDTH,HEIGHT));
+
+    while(!shouldQuit) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(5));
+        std::lock_guard<std::mutex> lock(mutex);
+        cam->readFrame(img);
+
+    }
+
+
+}
+
+void saveThread(cv::Mat img, std::string fileName) {
+    cv::imwrite(fileName, img);
```

Comment:
Again I think we should drop this thread, but this can also be done later.

``cv::imwrite()`` doesn't throw an error if it fails to save the file, it returns ``false`` instead. So I think we should check for this and print a warning if it hasn't saved properly.

Reply:
@tenxlenx I think this change is worth implementing before we go out (in case there's an error writing to disk).

Replace with:
``c++
if (!cv::imwrite(fileName, img))) {
    LOGW << "Could not save image file";
}
``